src/main/java/com/iexec/worker/chain/EnclaveAuthorizationService.java

```diff
@@ -28,21 +31,40 @@
 @Service
 public class EnclaveAuthorizationService {
 
-    public boolean isVerifiedEnclaveSignature(String chainTaskId, String resultHash, String resultSeal,
-                                              String enclaveSignature, String enclaveChallenge) {
-        if (enclaveChallenge == null || enclaveChallenge.isEmpty()) {
-            log.error("Cant verify enclave signature (enclave challenge not found) [chainTaskId:{}]", chainTaskId);
+    public boolean isVerifiedEnclaveSignature(String chainTaskId,
+                                              String resultHash,
+                                              String resultSeal,
+                                              String enclaveSignature,
+                                              String enclaveChallenge) {
+        String baseErrorMessage =
+                "Cannot verify enclave signature [chainTaskId:{}, ";
+        if (!BytesUtils.isNonEmptyBytes32(resultHash)) {
```

Comment:
I think it would be slightly better to add a descriptive message explaining why the checks have failed. Something like `resultHash is empty` or `enclaveSIgnature is not a valid signature`. That would probably speed up a bit log analyzing.

Reply:
Updated thanks :)